### PR TITLE
Let FlutterFragment not pop the whole activity by default when more fragments are in the activity

### PR DESCRIPTION
**Pull request by xster**
Fixes https://github.com/flutter/flutter/issues/69879

Addresses 4-6 of https://github.com/flutter/flutter/issues/67011#issuecomment-730665542. 

Custom implementations that does not want the default behavior, such as passing through the OnBackPressedDispatcher can override the popSystemNavigator method on FlutterFragment or FlutterActivity to have a custom behavior. 

cc @math1man

### CODE REVIEW

**Comment by xster** [comment]
Btw, that means the next g3 roll will break your app unless you implement the FlutterFragment override @math1man. Is that acceptable?

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java [hunk 1]
```diff
@@ -1032,6 +1028,12 @@ public boolean shouldRestoreAndSaveState() {
     return true;
   }
 
+  @Override
+  public boolean popSystemNavigator() {
```
Comment (hamdikahloun) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:1032]: Is this also available with  `FlutterActivity.createDefaultIntent(currentActivity)` and `FlutterFragment.createDefault()` ? if not maybe we should provide `PlatformPluginDelegate setter` something like flutterActivity.setPlatformPluginDelegate(new PlatformPluginDelegate).

Comment (xster) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:1032]: Generally, we prefer users to subclass then use one of the constructors like NewEngineIntentBuilder, CachedEngineIntentBuilder, NewEngineFragmentBuilder etc to build the subclass.

**Comment by hamdikahloun** [comment]
LGTM

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java (hunk 1 above)
Comment (hamdikahloun) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:1032]: Great work

**Review by hamdikahloun — APPROVED** [review]

File: shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java [hunk 2]
```diff
@@ -133,4 +139,86 @@ public void setNavigationBarDividerColor() {
...
+    PlatformPluginDelegate mockPlatformPluginDelegate = mock(PlatformPluginDelegate.class);
+    when(mockPlatformPluginDelegate.popSystemNavigator()).thenReturn(false);
+    PlatformPlugin platformPlugin =
+        new PlatformPlugin(mockFragmentActivity, mockPlatformChannel, mockPlatformPluginDelegate);
+
+    platformPlugin.mPlatformMessageHandler.popSystemNavigator();
+
+    verify(mockFragmentActivity, never()).finish();
+    verify(mockPlatformPluginDelegate, times(1)).popSystemNavigator();
+    verify(mockFragmentActivity, times(1)).getOnBackPressedDispatcher();
```
Comment (math1man) [inline shell/platform/android/test/io/flutter/plugin/platform/PlatformPluginTest.java:189]: Maybe also `verify(onBackPressedDispatcher, times(1)).onBackPressed();`?

**Comment by math1man** [comment]
Looks great, thanks!

@xster yes I'm fine with a temporary local breakage. We can ideally fix the issue in the drop, or just fix it immediately after.

**Comment by xster** [comment]
@math1man please watch the next roll to apply your local fix.

**Comment by math1man** [comment]
Will do.

**Pull request merged**